### ISSUE TITLE
Problem Installing gem

### ISSUE THREAD

**Issue by baob** [RadiusNetworks/scanbeacon-gem#28]
HI,

I tried a straight install as per the readme but I got a compilation error. `core_bluetooth.m:7:9: fatal error: 'CoreBluetooth/CoreBluetooth.h' file not found`

I've googled a bit and there are hints that 'CoreBluetooth seem to only be includable through IOBluetooth'  ( http://lists.apple.com/archives/bluetooth-dev/2013/May/msg00011.html ) but I've been unable to figure out how to apply this to the install.

Can you help at all ?

Thanks

**Comment by syoder** [comment]
What version of Mac OS do you have?  Does your laptop support BLE (ie, Bluetooth 4.0)?  Some older laptops supported an older version of Bluetooth that doesn't include Bluetooth Low Energy.

If your laptop does support BLE, there is one additional thing thing we can try: commenting out the `CoreBluetooth` import since we already import `IOBluetooth` which probably gets everything anyway.  I'm just not sure why the current code works for me but not for you. I'll see if I can do some testing on this and push a quick change sometime today.

**Comment by baob** [comment]
Hi, thanks for the response.

My Mac is on Mavericks 10.9.5 and it's hardware should support Bluetooth LE (Hardware LMP version -> 0x6 -> Bluetooth 4.0 -> Bluetooth LE according to this http://www.imore.com/how-tell-if-your-mac-has-bluetooth-40 )

Tried to discover my version of xcode but somehow that's triggered an xcode update. So that version info is lost but maybe the update will make a difference ?  Now on xcode 6.2 (6C131e). 

No ... still have the same compile issue.

**Comment by syoder** [comment]
I don't have a Mavericks machine to test on.  Would you be able to clone down the repo, and comment out a line and try a build?

Here's the line you'd need to comment out: https://github.com/RadiusNetworks/scanbeacon-gem/blob/master/ext/core_bluetooth/core_bluetooth.m#L7

I think once you clone it you'd need to do:
```bash
bundle
rake compile
rake install
```

If commenting out that line fixes it for you, I'll make that change and push out a new release.

**Comment by baob** [comment]
I have a problem at `rake compile`. I think the key error is `ld: framework not found CoreBluetooth`.  

The full set of messages is :
```
$ rake compile
cd tmp/x86_64-darwin14/core_bluetooth/2.2.3
/usr/bin/make
compiling ../../../../ext/core_bluetooth/core_bluetooth.m
../../../../ext/core_bluetooth/core_bluetooth.m:27:243: warning: declaration of 'struct BluetoothDeviceAddress' will not be visible outside of this
      function [-Wvisibility]
  ...advertisingType:(unsigned char)arg3 ownAddressType:(unsigned char)arg4 directAddressType:(unsigned char)arg5 directAddress:(struct BluetoothDevic...
                                                                                                                                        ^
../../../../ext/core_bluetooth/core_bluetooth.m:30:27: warning: declaration of 'struct BluetoothDeviceAddress' will not be visible outside of this
      function [-Wvisibility]
- (int)getAddress:(struct BluetoothDeviceAddress { unsigned char x1[6]; }*)arg1;
                          ^
2 warnings generated.
linking shared-object scan_beacon/core_bluetooth.bundle
ld: warning: directory not found for option '-L/Users/haven/.sm/pkg/active/lib'
ld: framework not found CoreBluetooth
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [core_bluetooth.bundle] Error 1
rake aborted!
Command failed with status (2): [/usr/bin/make...]
/Users/andy/.rvm/gems/ruby-2.2.3/gems/rake-compiler-0.9.9/lib/rake/extensiontask.rb:157:in `block (2 levels) in define_compile_tasks'
/Users/andy/.rvm/gems/ruby-2.2.3/gems/rake-compiler-0.9.9/lib/rake/extensiontask.rb:156:in `block in define_compile_tasks'
/Users/andy/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
/Users/andy/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => compile => compile:x86_64-darwin14 => compile:core_bluetooth:x86_64-darwin14 => copy:core_bluetooth:x86_64-darwin14:2.2.3 => tmp/x86_64-darwin14/core_bluetooth/2.2.3/core_bluetooth.bundle
(See full trace by running task with --trace)
```

**Comment by syoder** [comment]
ah, you may need to comment out this line as well: https://github.com/RadiusNetworks/scanbeacon-gem/blob/master/ext/core_bluetooth/extconf.rb#L12

**Comment by baob** [comment]
Great !  That now compiles and the specs run. Haven't yet detected a beacon so you may want to hold off merging ?

FWIW this is my branch:  https://github.com/baob/scanbeacon-gem/tree/fix-compile-error

**Comment by syoder** [comment]
Feel free to submit a pull request (or I can add the changes directly if you prefer).  I'll test here and make sure that it works before merging.

**Comment by syoder** [comment]
In my testing, not linking to CoreBluetooth didn't affect any functionality.  I think IOBluetooth includes everything you need.

I've released v0.7.8 to rubygems, which includes this fix.  Please let me if this works, or reopen this issue if it doesn't.

**Comment by baob** [comment]
Thanks. I can confirm I can pull down v0.7.8 from rubygems and it compiles cleanly on Mavericks. I can also run the detection loop code in IRB (but have not detected beacons which was what held me up from giving a definitive reply before).
